Donation-platform/src/main/java/com/google/DbCalls.java

```diff
@@ -13,200 +14,281 @@
-    }
-    return tagDataStore;
-  }
-  // Function that returns user from user name.
-  public Users getUserByUserName(String userName) throws Exception {
-    Query query =
-        new Query(USERS).setFilter(new FilterPredicate(USERNAME, FilterOperator.EQUAL, userName));
-    return setUsersClass(datastore.prepare(query).asSingleEntity());
-  }
-  // Function that returns user from email.
-  public Users getUserByEmail(String email) throws Exception {
-    Query query =
-        new Query(USERS).setFilter(new FilterPredicate(EMAIL, FilterOperator.EQUAL, email));
-    return setUsersClass(datastore.prepare(query).asSingleEntity());
-  }
+        return charityDataStore;
+    }
+    // Function returns all the tag objects in the database.
+    public Collection<Tag> getAllTags() throws Exception
+    {
+        ArrayList< Tag > tagDataStore = new ArrayList < Tag > ();
+        for (Entity entity : datastore.prepare(new Query(TAG)).asIterable())
+        {
+            tagDataStore.add(setTagClass(entity));
+        }
+        return tagDataStore;
+    }
+    // Function returns all the users in the database.
+    public Collection<Users> getAllUsers() throws Exception
+    {
+        ArrayList< Users > userDataStore = new ArrayList < Users > ();
+        for (Entity entity : datastore.prepare(new Query(USERS)).asIterable())
+        {
+            userDataStore.add(setUsersClass(entity));
+        }
+        return userDataStore;
+    }
+    // Function adds charity to database.
+    // Could be used in conjunction with user charity add form.
+    public void addCharity(String name, String link, String imgsrc, Collection<Tag>  categories, String description) throws Exception
+    {
+        Entity charityEntity = new Entity(CHARITY);
+        charityEntity.setProperty(NAME, name);
+        charityEntity.setProperty(LINK, link);
+        charityEntity.setProperty(IMGSRC, imgsrc);
+        charityEntity.setProperty(CATEGORIES, getTagIdsByObject(categories));
+        charityEntity.setProperty(DESCRIPTION, description);
+        charityEntity.setProperty(TRENDING_SCORE, 0.0);
+        charityEntity.setProperty(USER_RATING, 0.0);
+        datastore.put(charityEntity);
+    }
+    // Function adds tag to the database.
+    public void addTag(String name, double trendingScore) throws Exception
+    {
+        Entity tagEntity = new Entity(TAG);
+        tagEntity.setProperty(NAME, name);
+        tagEntity.setProperty(TRENDING_SCORE, trendingScore);
+        datastore.put(tagEntity);
+    }
+    // Function adds user to the database.
+    public void addUser(String userId, String userName, String email,
+                        Collection<Tag> userInterests, Collection<Charity>  charitiesDonatedTo) throws Exception
+    {
+        Entity userEntity = new Entity(USERS, userId);
+        userEntity.setProperty(USERNAME, userName);
+        userEntity.setProperty(EMAIL, email);
+        userEntity.setProperty(USER_INTERESTS, userInterests);
```

Comment:
I dont see modified test file, do you think test are going to fail with this change?

Reply:
The test file is here. Just scroll down and click load diff.

Reply:
got it, my bad